lib/heroku/command/domains.rb

```diff
@@ -13,17 +14,35 @@ class Domains < Base
     #Examples:
     #
     # $ heroku domains
-    # === Domain names for example
-    # example.com
+    # === Development Domain
+    # example.herokuapp.com
+    #
+    # === Custom Domains
+    # Domain Name  CNAME Target
```

Comment:
I think 'DNS Record Target' would capture everyone's concerns. It works for both CNAME and ALIAS, speaks to the right hand side of the record, and hints that it is for DNS. If we change it here, we also need to update Dashboard.